staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -43,7 +43,7 @@ func HTTPWrappersForConfig(config *Config, rt http.RoundTripper) (http.RoundTrip
 		rt = config.WrapTransport(rt)
 	}
 
-	rt = DebugWrappers(rt)
+	rt = MaybeDebugWrappers(rt, config.ForceDebugWrappers)
```

Comment:
@alvaroaleman, @ahmetb: which APIs built on top of his would need to be extended to make this useful for consumers of controller-runtime?

Reply:
I think the `Maybe` portion is just not gonna be easy to work with. We want this to be ideally enabled for all controllers by default, and most controllers explicitly provide a `rest.Config` to the controller-runtime, and they won't do `ForceDebugWrappers=true`.

I'd really want us to reconsider always wrapping. I remember the concern was having another point where ctx cancellation could happen, or another method shows up in the call stack? These don't strike me as too critical, and they can be mitigated to a good extent.